### PR TITLE
feat: Rfq healtcheck worker

### PR DESCRIPTION
**Pull request by PirosB3**
Adds a small function called `isWorkerReadyAndAbleAsync()` which allows us to make sure a RFQM worker is ready before picking up work

### CODE REVIEW

File: src/runners/rfqm_meta_transaction_consumer_runner.ts [hunk 1]
```diff
@@ -65,6 +68,72 @@ if (require.main === module) {
...
+            },
+            'Worker does not have enough balance to trade.',
+        );
+    }
+    return hasEnoughBalance;
+}
+
+async function workerHasNoPendingTransactionsAsync(accountAddress: string, wrapper: Web3Wrapper): Promise<boolean> {
+    const lastNonceOnChain = await wrapper.getAccountNonceAsync(accountAddress);
+    const lastNoncePending = await wrapper.getAccountNonceAsync(accountAddress, BlockParamLiteral.Pending);
```
Comment (phil-ociraptor) [inline src/runners/rfqm_meta_transaction_consumer_runner.ts:101]: this is cool! Didn't know you could do this

File: src/runners/rfqm_meta_transaction_consumer_runner.ts [hunk 2]
```diff
@@ -65,6 +68,72 @@ if (require.main === module) {
...
 }
 
+/**
+ * Returns true if the worker has a sufficient balance to trade
+ * @param accountAddress the EOA address
+ * @param accountBalance the EOA address balance
+ * @param gasPriceBaseUnits fast gas price (wei)
+ * @returns true if the worker has sufficient balance
+ */
+export function workerHasEnoughBalance(
```
Comment (phil-ociraptor) [inline src/runners/rfqm_meta_transaction_consumer_runner.ts:78]: these are awesome! could we put `workerHasEnoughBalance` `workerHasNoPendingTransactionsAsync` and `isWorkerReadyAndAbleAsync` in a new file called `rfqm_worker_balance_utils` or something? Then we can change the merge base to `master` and get this checked in without being blocked by the other commit

Comment (PirosB3) [reply src/runners/rfqm_meta_transaction_consumer_runner.ts]: Done!

**Comment by PirosB3** [comment]
@phil-ociraptor I think https://github.com/0xProject/0x-api/pull/588 is cleaner

**Review by PirosB3 — COMMENTED** [review]
A few minor comments! @phil-ociraptor for some reason GH is preventing me tfrom approvcing this PR

**Pull request closed without merging**